source/common/ssl/context_config_impl.cc

```diff
@@ -74,8 +75,10 @@ ContextConfigImpl::ContextConfigImpl(const envoy::api::v2::auth::CommonTlsContex
     }
     if (!verify_subject_alt_name_list_.empty()) {
       throw EnvoyException(fmt::format("SAN-based verification of peer certificates without "
-                                       "trusted CA is insecure and not allowed",
-                                       certificateRevocationListPath()));
+                                       "trusted CA is insecure and not allowed"));
+    }
+    if (allow_expired_certificate_) {
+      throw EnvoyException(fmt::format("Cannot ignore certificate expiration without trusted CA"));
```

Comment:
Maybe something like `"Certificate validity period is always ignored without trusted CA"` would be a bit more accurate? What do you think?

Reply:
Works for me.